### PR TITLE
Add HttpClient guidance

### LINKED ISSUE #26510: HttpClient - items to improve
- [ ] Add cross-link to https://docs.microsoft.com/en-us/dotnet/standard/microservices-architecture/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests
- [ ] Add examples of using Polly and [Polly.Extensions.Http](https://www.nuget.org/packages/Polly.Extensions.Http/) nuget PackageReference to a project. These are "opinionated" libraries to assist in transient fault handling.
- [ ] Incorporate @karelz feedback:
    > Unfortunately, this recommendation is not complete.
    > We need to warn users also about having long-existing HttpClient instances as that may lead to stale DNS usage. That can lead to non-functional connections.
    > 
    > The recommendation should be either to:
    > 
    > * Use `HttpClient` instances as statics, which are recycled on regular basis, or
    > * Use `HttpClientFactory` which works around this problem, or
    > * Use static instances on .NET Core while setting `System.Net.Http.SocketsHttpHandler.PooledConnectionLifetime`.
- [ ] Regarding PooledConnectionLifetime, consider adding reference to https://github.com/dotnet/corefx/pull/29822 which I believe fixes this in .NET Core 2.2 (Stephen Toub's commit is on May 22, 2018 and .NET Core 2.1 was released on May 30th, 2018).  In other words, if your application needs to target .NET Core 2.1, this approach can probably run into problems described in that PR.
- [ ] Consider describing the benefits of using HttpClient via HttpClientFactory, including remarks in SockerHttpHandler https://docs.microsoft.com/en-us/dotnet/api/system.net.http.socketshttphandler?view=netcore-3.1#remarks

_Originally posted by @jzabroski in https://github.com/dotnet/dotnet-api-docs/pull/2315#issuecomment-587119488_

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #26510.

[Preview link](https://review.docs.microsoft.com/en-us/dotnet/fundamentals/networking/httpclient-guidelines?branch=pr-en-us-29568).

### CODE REVIEW

**Review by tdykstra — APPROVED** [review]
LGTM

File: docs/fundamentals/networking/httpclient-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+title: HTTPClient guidelines for .NET
```
Comment (IEvangelist) [inline docs/fundamentals/networking/httpclient-guidelines.md:2]:
```suggestion
title: HttpClient guidelines for .NET
```

File: docs/fundamentals/networking/httpclient-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+description: Learn about using HttpClient instances to send HTTP requests and how you can manage clients using IHttpClientFactory in your .NET apps.
+author: gewarren
+ms.author: gewarren
+ms.date: 05/19/2022
+---
+# Guidelines for using HTTPClient
```
Comment (IEvangelist) [inline docs/fundamentals/networking/httpclient-guidelines.md:8]:
```suggestion
# Guidelines for using HttpClient
```

**Review by IEvangelist — APPROVED** [review]
Looks good to me, :shipit: when you're ready. There are two suggestions I'd like to see incorporated before this is merged.

**Comment by Henderv** [comment]
> Fixes #26510.
> 
> [Preview link](https://review.docs.microsoft.com/en-us/dotnet/fundamentals/networking/httpclient-guidelines?branch=pr-en-us-29568).

**Pull request merged**